portal_tables/unify_grant_tables.py

```diff
@@ -0,0 +1,145 @@
+consistent in the CCKP admin project table.
+
+author: victor.baham
+"""
+
+# command line arguments: 1. Portal - Grants Merged Synapse ID 2. CCKP Admin Project Synapse ID
+# test value for argument 1: syn42801895
+# test value for argument 2: syn35629947 
+
+import argparse
+import synapseclient
+from synapseclient import Project, File, Folder
+from synapseclient import Schema, Column, Table, Row, RowSet, as_table_columns
+import itertools
+import pandas as pd
+
+def get_args():
+    #Retrieve Synapse table IDs from command line
+    parser = argparse.ArgumentParser(
+        description="Merge publications, datasets, and tools from Grants Merged table into CCKP admin table")
+    parser.add_argument("-p1", "--project1",
+                        type=str, required=True,
+                        help="Portal - Grants Merged Synapse ID.")
+    parser.add_argument("-p2", "--project2",
+                        type=str, required=True,
+                        help=("CCKP - MC2 Admin Synapse ID"))
+    parser.add_argument("--dryrun", action="store_true")
+    return parser.parse_args()
+
+def getManifestsFromGrant(syn,theParent):
+    """This function examines the GrantView table
+    and returns a list of all Synapse table IDs
+    separated by manifest type (publications, datasets,
+    tools) for each grant"""
+
+    # Access each grant project by accessing 'grantId'
+    # column in grantView manifest
+    # synId for grantView is hardcoded in and
+    # will need to be changed for future use
+    df = syn.tableQuery(f"select {'grantId'} from {theParent}").asDataFrame()
+    gIds = df['grantId'].tolist()
+
+    # for each grant, get publications,
+    # datasets, and tools 
+
+    # value inside dictionary will reset upon each
+    # loop iteration, so appending to this list 
+    # maintains each one
+    mTypes = []
+    mIds = []
+    for gId in gIds:
+        grantX = syn.get(gId)
+        # By accessing the grants, it is possible
+        # to access their 'children'
+
+        # extract manifest type and table id from generator
+        mTypes.append([m['name'] for m in syn.getChildren(grantX, includeTypes=['table'])])
+        mIds.append([m['id'] for m in syn.getChildren(grantX, includeTypes=['table'])])
+
+    # with each synapse table id mapped to a type of manifest
+    # it is possible to begin adding them to the staging table
+    # but should flatten list first for easier key-val mapping
+    flat_mTypes = list(itertools.chain(*mTypes))
+    flat_mIds = list(itertools.chain(*mIds))
+
+    idToManType = dict(zip(flat_mIds, flat_mTypes))
+    print(idToManType)
+
+    # use dictionary of synapse id to manifest type
+    # to prepare to separate data into rows
+    pubsId = []
+    datasetsId = []
+    toolsId = []
+
+    for key,val in idToManType.items():
+        if val == 'publications':
+            pubsId.append(key)
+        elif val == 'datasets':
+            datasetsId.append(key)
+        else:
```

Comment:
Careful with this one.  This assumes that if the table isn't `publications` or `datasets`, then it'll always be `tools`.  But what happens if the centers create a table on their own into the Synapse project?

Reply:
This also assumes that the table names are going to be `publications` and/or `datasets`.  But what if they're named `PublicationView` or `Publications` etc.

Something to think about.

Reply:
I wonder if @MasterPastures could use `in`. E.g. `elif 'dataset' in val.lower()` or something like that?

Reply:
This would account for capitalization differences and plurals